okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java

```diff
@@ -396,6 +403,55 @@ public OkHttpChannelBuilder connectionSpec(
     return this;
   }
 
+  /**
+   * Sets the connection specification used for secure connections.
+   *
+   * <p>By default a modern, HTTP/2-compatible spec will be used.
+   *
+   * <p>This method is only used when building a secure connection. For plaintext
+   * connection, use {@link #usePlaintext()} instead.
+   *
+   * @param tlsVersions List of tls versions.
+   * @param cipherSuites List of cipher suites.
+   * @param supportsTlsExtensions Support of TLS extensions.
+   */
+  public OkHttpChannelBuilder tlsConnectionSpec(
+          String[] tlsVersions, String[] cipherSuites,
+          boolean supportsTlsExtensions) {
+    Preconditions.checkState(!freezeSecurityConfiguration,
+            "Cannot change security when using ChannelCredentials");
+    Preconditions.checkNotNull(tlsVersions, "tls versions must not null");
+    Preconditions.checkNotNull(cipherSuites, "ciphers must not null");
+
+    this.connectionSpec = new ConnectionSpec.Builder(true)
+            .supportsTlsExtensions(supportsTlsExtensions)
+            .tlsVersions(tlsVersions)
+            .cipherSuites(cipherSuites)
+            .build();
+    return this;
+  }
+
+  /**
+   * For secure connection, provides a ConnectionSpec to specify Cipher suite and
+   * TLS versions.
+   *
+   * <p>By default a modern, HTTP/2-compatible spec will be used.
+   *
+   * <p>This method is only used when building a secure connection. For plaintext
+   * connection, use {@link #usePlaintext()} instead.
+   *
+   * @throws IllegalArgumentException
+   *         If {@code connectionSpec} is not with TLS
+   */
+  public OkHttpChannelBuilder internalConnectionSpec(
```

Comment:
I don't think we should add this method, as it requires using an internal class which is a no-no.

My real plan/expectation is to add okhttp3 ConnectionSpec support, powered by another implementation of `Utils.convertSpec()`.

Reply:
Yeah, not having it now is what I was expecting. The String-based version looks like it should be sufficient for you, so this method isn't strictly necessary. Okhttp3 stuff will happen later.

Reply:
> just support both major versions in parallel

FWIW, my plan was just to support both in parallel. But I don't think you need to take that on; this PR is plenty helpful already without the feature creep.

Reply:
If I find some time in the next weeks I could prepare a PR